lib/file_name.rb

```diff
@@ -0,0 +1,82 @@
+# 1
+puts "Which method would you like to use? (add, subtract, multiply, or divide)"
+methods = gets.chomp
+
+puts "You chose " + methods
+
+puts "What is the 1st number?"
+num1 = gets.chomp
+puts "You entered " + num1
+
+puts "What is the 2nd number?"
+num2 = gets.chomp
+puts "You entered " + num2
+
+if methods == "add"
+  puts num1.to_i + num2.to_i
+elsif methods == "subtract"
+  puts num1.to_i - num2.to_i
+elsif methods == "multiply"
+  puts num1.to_i * num2.to_i
+elsif methods == "divide"
+  puts num1.to_i / num2.to_i
+else
+  puts "Enter Valid method"
+end
+
+# 2
+def reverseStr(str)
+  newArray = []
+  revstr = str.split('')
+  for i in (0...revstr.length)
+    puts revstr.pop()
+    newArray.push(revstr)
+  end
+  puts newArray
+end
+
+reverseStr('DedElectric')
+
```

Comment:
Alternative - 
puts  'Enter a string:'
 reverse = gets.chomp.strip
 
 arr = reverse.chars
 
 for i in (0..(arr.length-1)/2)
   arr[arr.length-1-i], arr[i] = arr[i], arr[arr.length-1-i]
 end
 
 puts arr.inspect